Assets/_Scripts/Actions/ActionBase.cs

```diff
@@ -49,11 +49,37 @@ public abstract class ActionBase : ScriptableObject {
             potentialTargets = targetIdType == TargetType.MY_TEAM ? players : enemies;
         }
 
+        return potentialTargets;
+    }
+
+    public List<FightingEntity> GetTargets(FightingEntity user, List<int> targetIds){ 
+        List<FightingEntity> potentialTargets = GetPotentialTargets(user);
         List<FightingEntity> targets = new List<FightingEntity>();
         foreach (int target in targetIds) {
-            targets.Add(potentialTargets[target]);
+            if (target < potentialTargets.Count) {
+                targets.Add(potentialTargets[target]);
+            }
         }
 
         return targets;
     }
+
+    public List<FightingEntity> GetTargets(FightingEntity user, int targetId) {
+        return this.GetTargets(user, new List<int>{targetId});
+    }
+
+    protected int GetTargetIdFromString(string str, FightingEntity user) {
+        int targetId;
+        if (!int.TryParse(str, out targetId)) {
```

Comment:
As a future suggestion, we should update ActionBase to implement a `validate command` method. This way we don't have to clutter the GetTarget function with parsing checks like these.

Reply:
We do have "BasicValidation" in ActionBase, but the index string can be in many different spots (.e.g in attack "!a 2" it is in index 1, whereas in magic "!m fire 2" it is in index 2

Reply:
ok - this can be done later but should probably be a virtual function.